database/DatabaseExampleSwiftUI/DatabaseExample/Shared/Views/TopPostsView.swift

```diff
@@ -18,16 +18,36 @@ import SwiftUI
 import Firebase
 
 struct TopPostsView: View {
```

Comment:
How much different are `RecentPostsView` and `TopPostsView`? Can we have a generic `PostsView` that accepts a parameter like `type` that can be used instead of `tabOpened`?

Reply:
Do you mean renaming tabOpened to type or something else? RecentPostView and TopPostsView are retrieving data from two different parts of Firebase Realtime Database.

Reply:
My main question is if `RecentPostsView` and `TopPostsView` display a given list of posts anyhow different? It looks to me that no and the only difference is `tabOpened` value. If it is so, I don't think we need to have two different classes, we can just have a single one and initialize it with `tabOpened` value.

My second point is about `tabOpened` name itself. It seems not exactly precise to me. If we have a view for a list of posts I would name the parameter like `postType` or something along these lines. I'm not too strong about it though.

Reply:
I see did you mean combining all 3 `RecentPostsView`, `MyPostsView`, and `TopPostsView` into one file and pass in the tabOpened value? Currently in each view it contains its own `@StateObject` of type `UserViewModel` that refreshes on appear, it seems like the app is still functional after combining the 3 views into one file. So to clarify I will pass in the `tabOpened` and `navigation title` to a single class instead of having different classes?

Reply:
SGTM

Reply:
Sounds good. Will restructure on the next PR.